neurokit2/ppg/ppg_intervalrelated.py

```diff
@@ -97,27 +95,28 @@ def ppg_intervalrelated(data, sampling_rate=1000):
 # =============================================================================
 
 
-def _ppg_intervalrelated_formatinput(data, output={}):
-
+def _ppg_intervalrelated_formatinput(data: pd.DataFrame):
```

Comment:
is there a reason you added a type annotation here? I do think that type annotations are generally good to have but it's inconsistent with the rest of the code

Reply:
Looks good to me, but curious about that annotation as well.

In general yes, I also think type annotations are good, although NK might not be a priority repo to add them, as in general we are pretty flexible by design regarding the type of input (i.e., we try to accommodate potential user "mistakes")

Reply:
It is indeed more of a convenience for the "deep-diving" user - such that the employed IDE gets what the incoming data type is and thus enables the suggestion features for the variable. Regarding flexibility, the functions were before and still are only written to handle data as a pandas DataFrame as that is what is created from ppg_process beforehand.  

Me personally, I love using type hints as it makes using functions clearer and easier - and they also help with getting back into code that I did not work on for some time a lot faster - so maybe I simply added it from a force of habit as well.

If pure functionality and consistency with the rest of the code are the goals, the type hint can be removed without consequence. Let me know, if you want me to drop it.

Reply:
I don't have a big experience with type hints, but I can see their value. 

I guess either:
- we remove it here for consistency *for now* (and eventually someday working on an update by type hitting the whole codebase)
- We leave it and we implicitly start slowly type-hitting here and there whenever we do some code updates (i.e., the progress will be slow and random). 

@danibene what do you think?

Reply:
> I don't have a big experience with type hints, but I can see their value. 
> 
> I guess either:
> - we remove it here for consistency *for now* (and eventually someday working on an update by type hitting the whole codebase)
> - We leave it and we implicitly start slowly type-hitting here and there whenever we do some code updates (i.e., the progress will be slow and random). 
> 
> @danibene what do you think?
> 
> 

I also don't have much experience with type hints, but I'd err on the side of removing it for now.

One option to start with, to make sure we're not biting off more than we can chew, could be to create separate stub files like pandas https://github.com/pandas-dev/pandas-stubs

I recently started this on my own so I could use mypy in repositories using certain neurokit functions, if you'd like we could add this to the neuropsychology org and request that contributors of new functions contribute there too:
https://github.com/danibene/neurokit2-stubs/tree/dev

Reply:
> I also don't have much experience with type hints, but I'd err on the side of removing it for now.

Let's do that then

> could be to create separate stub files

Oh boy it's the first time I hear about stubs 😵‍💫 Let me do some reading first and then we can see what to do regarding the infrastructure